src/functional/ffront/foast_passes/type_deduction.py

```diff
@@ -409,6 +409,16 @@ def _deduce_binop_type(
         if node.op == ct.BinaryOperator.POW:
             return left_type
 
+        if node.op == ct.BinaryOperator.MOD and (
+            isinstance(right_type, ct.ScalarType) and not type_info.is_integral(right_type)
```

Comment:
I don't think the right hand side needs to be a scalar (if that is what you wanted to express, currently the conditions seem wrong).